server/crawler/index.ts

```diff
@@ -329,17 +333,16 @@ export async function getCralwerData(url: string) {
 }
 
 export async function crawlOgImage(url: string) {
-  const { data } = await axios.get(encodeURI(url as string))
-  const $ = cheerio.load(data)
+  const content = await getContentFromUrl(url)
+  const $ = cheerio.load(content)
   const metas = $('meta')
   let ogImageUrl = ''
-  for (let i = 0; i < Object.keys(metas).length; i += 1) {
-    const { attribs } = metas[i]; // keys are integers 1 to N
-    if (attribs?.property === 'og:image') {
-      ogImageUrl = attribs.content
-      break
+  Object.keys(metas).forEach((key: any) => {
+    const { property, content: value } = metas[key].attribs || {}
+    if (property === 'og:image') {
+      ogImageUrl = value
     }
```

Comment:
Actually the for loop version allows early return, but i guess unifying the way we write allow us to refactor later